servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -1783,6 +1806,166 @@ private void setStateFromEntry(SetStateEntry entry) throws FileDoesNotExistExcep
   }
 
   /**
+   * Sets the acl of a file or directory.
+   *
+   * @param path to be set acl on
+   * @param options acl option to be set
+   * @return true if set successfully, false otherwise
+   * @throws TachyonException
+   */
+  public boolean setAcl(TachyonURI path, SetAclOptions options) throws TachyonException {
+    if (options.getOwner() != null) {
+      return setOwner(path, options.getOwner(), options.isRecursive());
+    }
+
+    if (options.getGroup() != null) {
+      return setGroup(path, options.getGroup(), options.isRecursive());
+    }
+
+    if (options.getPermission() != -1) {
+      return setPermission(path, options.getPermission(), options.isRecursive());
+    }
+    return false;
+  }
+
+  /**
+   * Sets the user to be the owner of the path. Only a super user can change the owner of a path.
+   *
+   * @param path to be set owner on
+   * @param user to be set as the owner
+   * @param recursive indicates whether to set all the children recursively when path is a dir
+   * @return true if set successfully, false otherwise
+   * @throws TachyonException
+   */
+  private boolean setOwner(TachyonURI path, String user, boolean recursive)
+      throws TachyonException {
+    FileSystemPermissionChecker.checkSuperuser(getClientUser(), getGroups(getClientUser()));
+    synchronized (mInodeTree) {
+      long opTimeMs = System.currentTimeMillis();
+      Inode targetInode = mInodeTree.getInodeByPath(path);
+      if (recursive && targetInode.isDirectory()) {
+        List<Inode> inodeChildren =
+            mInodeTree.getInodeChildrenRecursive((InodeDirectory) targetInode);
+        for (Inode inode : inodeChildren) {
+          setAclInternal(inode, opTimeMs, user, null, (short) -1);
+        }
+      }
+      setAclInternal(targetInode, opTimeMs, user, null, (short) -1);
+    }
+    return true;
+  }
+
+  /**
+   * Sets the group of the path. Only a super user or the path owner can change.
+   *
+   * @param path to be set group on
+   * @param group to be set
+   * @param recursive indicates whether to set all the children recursively when path is a dir
+   * @return true if set successfully, false otherwise
+   * @throws TachyonException
+   */
+  private boolean setGroup(TachyonURI path, String group, boolean recursive) throws
+      TachyonException {
+    synchronized (mInodeTree) {
+      checkOwner(path);
+      long opTimeMs = System.currentTimeMillis();
+      Inode targetInode = mInodeTree.getInodeByPath(path);
+      if (recursive && targetInode.isDirectory()) {
+        List<Inode> inodeChildren =
+            mInodeTree.getInodeChildrenRecursive((InodeDirectory) targetInode);
+        for (Inode inode : inodeChildren) {
+          checkOwner(mInodeTree.getPath(inode));
+        }
+        for (Inode inode : inodeChildren) {
+          setAclInternal(inode, opTimeMs, null, group, (short) -1);
```

Comment:
indicates permission does not take effect in the function. 
Update the methods to be simpler in latest update.